tests/std/tests/P1502R1_standard_library_header_units/test.cpp

```diff
@@ -0,0 +1,973 @@
+            caught_bad_alloc = true;
+        } catch (...) {
+            assert(false);
+        }
+
+        assert(caught_bad_alloc);
+
+        const int* const ptr = new (nothrow) int{1729};
+        assert(ptr);
+        assert(*ptr == 1729);
+        delete ptr;
+
+        static_assert(hardware_constructive_interference_size == 64); // implementation-defined
+        static_assert(hardware_destructive_interference_size == 64); // implementation-defined
+    }
+
+    {
+        puts("Testing <numbers>.");
+        static_assert(3.14 < numbers::pi && numbers::pi < 3.15);
+        static_assert(2.71828f < numbers::e_v<float> && numbers::e_v<float> < 2.71829f);
+    }
+
+    {
+        puts("Testing <numeric>.");
+        constexpr int arr[]{1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+        assert(accumulate(begin(arr), end(arr), 0) == 55);
+        static_assert(accumulate(begin(arr), end(arr), 0) == 55);
+    }
+
+    {
+        puts("Testing <optional>.");
+#ifndef MSVC_INTERNAL_TESTING // TRANSITION, VSO-1236047 (c1xx!FindConversionFunctions() assertion)
+        constexpr optional<int> opt{in_place, 1729};
+        assert(opt.has_value());
+        assert(opt.value() == 1729);
+        static_assert(opt.has_value());
+        static_assert(opt.value() == 1729);
+
+        constexpr optional<int> empty{nullopt};
+        assert(!empty.has_value());
+        assert(empty.value_or(-1) == -1);
+        static_assert(!empty.has_value());
+        static_assert(empty.value_or(-1) == -1);
+#endif // ^^^ no workaround ^^^
+    }
+
+    {
+        puts("Testing <ostream>.");
+        const ostream os{nullptr};
+        assert(os.rdbuf() == nullptr);
+    }
+
+    {
+        puts("Testing <queue>.");
+#ifdef MSVC_INTERNAL_TESTING // TRANSITION, DevCom-1160260 (partial specialization)
+        queue<int> q;
+        q.push(10);
+        q.push(20);
+        q.push(30);
+        assert(q.size() == 3);
+        assert(q.front() == 10);
+        q.pop();
+        assert(q.front() == 20);
+        q.pop();
+        assert(q.front() == 30);
+        q.pop();
+        assert(q.empty());
+#endif // ^^^ no workaround ^^^
+
+        priority_queue<int> pq;
+        pq.push(50);
+        pq.push(10);
+        pq.push(1729);
+        pq.push(200);
+        assert(pq.size() == 4);
+        assert(pq.top() == 1729);
+        pq.pop();
+        assert(pq.top() == 200);
+        pq.pop();
+        assert(pq.top() == 50);
+        pq.pop();
+        assert(pq.top() == 10);
+        pq.pop();
+        assert(pq.empty());
+    }
+
+    {
+        puts("Testing <random>.");
+        minstd_rand0 lcg;
+
+        for (int i = 0; i < 9999; ++i) {
+            (void) lcg();
+        }
+
+        assert(lcg() == 1043618065); // N4861 [rand.predef]/1
+    }
+
+    {
+        puts("Testing <ranges>.");
+        constexpr int arr[]{11, 0, 22, 0, 33, 0, 44, 0, 55};
+        assert(ranges::count(arr, 0) == 4);
+        static_assert(ranges::count(arr, 0) == 4);

```

Comment:
This is testing `<algorithm>`. Might I suggest instead:
```suggestion
        assert(ranges::distance(views::filter(arr, [](int x) { return x == 0; })) == 4);
        static_assert(ranges::distance(views::filter(arr, [](int x) { return x != 0; })) == 5);
```

Reply:
Done, with modifications (I moved `ranges::count` over to the `<algorithm>` section, and had to use `ref_view` to completely avoid CTAD).